compiler/ballerina-parser/src/main/java/io/ballerina/compiler/internal/parser/AbstractLexer.java

```diff
@@ -134,4 +134,117 @@ protected STNode getLeadingTrivia() {
         this.leadingTriviaList = new ArrayList<>(INITIAL_TRIVIA_CAPACITY);
         return trivia;
     }
+
+    /**
+     * Check whether a given char is an identifier following char.
+     * <p>
+     * <code>IdentifierFollowingChar := IdentifierInitialChar | Digit</code>
+     *
+     * @param c character to check
+     * @return <code>true</code>, if the character is an identifier following char. <code>false</code> otherwise.
+     */
+    protected static boolean isIdentifierFollowingChar(int c) {
+        return isIdentifierInitialChar(c) || isDigit(c);
+    }
+
+    /**
+     * Check whether a given char is a digit.
+     * <p>
+     * <code>Digit := 0..9</code>
+     *
+     * @param c character to check
+     * @return <code>true</code>, if the character represents a digit. <code>false</code> otherwise.
+     */
+    protected static boolean isDigit(int c) {
+        return ('0' <= c && c <= '9');
+    }
+
+    /**
+     * Check whether a given char is a hexa digit.
+     * <p>
+     * <code>HexDigit := Digit | a .. f | A .. F</code>
+     *
+     * @param c character to check
+     * @return <code>true</code>, if the character represents a hex digit. <code>false</code> otherwise.
+     */
+    protected static boolean isHexDigit(int c) {
+        if ('a' <= c && c <= 'f') {
+            return true;
+        }
+        if ('A' <= c && c <= 'F') {
+            return true;
+        }
+        return isDigit(c);
+    }
+
+    /**
+     * Check whether a given char is an identifier start char.
+     * <p>
+     * <code>IdentifierInitialChar := A .. Z | a .. z | _ | UnicodeIdentifierChar</code>
+     *
+     * @param c character to check
+     * @return <code>true</code>, if the character is an identifier start char. <code>false</code> otherwise.
+     */
+    protected static boolean isIdentifierInitialChar(int c) {
+        if ('A' <= c && c <= 'Z') {
+            return true;
+        }
+
```

Comment:
We can remove these new lines.

Reply:
I think leaving a newline after a block whenever appropriate is a good practice. Ref: https://github.com/ballerina-platform/ballerina-lang/pull/26717#discussion_r527414794 also.